### DISCUSSION TITLE
Configure clangd which was not installed via Mason

### DISCUSSION THREAD

**Discussion by bartekjaszczak** [VonHeikemen/lsp-zero.nvim#167] [Q&A] [answered]
Hi. I can't use clangd installed through Mason (it's clang 15.0.6), because of some missing dependencies (GLIBC_2.18). Because of that, I wanted to use clangd which is already installed on my system (15.0.5), but I can't really make it to work. I tried to set it up in several ways, one of which is the below (which actually used to work some time ago):

```lua
local lsp = require("lsp-zero")
lsp.preset("recommended")
lsp.ensure_installed({
  "lua_ls",
)}

local capabilities = vim.lsp.protocol.make_client_capabilities()
capabilities.offsetEncoding = { "utf-16" }

lsp.configure("clangd", {
  capabilities = capabilities,
  on_attach = function(client, bufnr)
    local opts = { buffer = bufnr, remap = false }

    vim.keymap.set("n", "<leader>vh", "<cmd>ClangdSwitchSourceHeader<CR>", opts)
  end
})

lsp.on_attach(function(client, bufnr)
  -- some keymappings here
end)

lsp.setup()
```
Full config can be found here: [repo](https://github.com/bartekprtc/nvim/blob/main/after/plugin/lsp.lua)

As a result, when I check LspInfo, only "lua_ls" shows in the Configured servers list. Whenever I open a cpp file, only one client attaches (null-ls).

clangd command is executable in the terminal. `:lua require('lsp-zero.check').run('clangd')` yields:

> LSP server: clangd
> - was not installed with mason.nvim
> - hasn't been configured with lspconfig

Has anyone had similar issues?

**Comment by VonHeikemen** [comment, marked as answer]
Add `force_setup = true` in the `.configure()` call.

```lua
lsp.configure("clangd", {
  force_setup = true,
  capabilities = capabilities,
  on_attach = function(client, bufnr)
    local opts = { buffer = bufnr, remap = false }

    vim.keymap.set("n", "<leader>vh", "<cmd>ClangdSwitchSourceHeader<CR>", opts)
  end
})
```

**Reply by bartekjaszczak** [reply]
Amazing, thank you!
